docs/tutorials/acme/dns-validation.rst

```diff
@@ -36,27 +36,30 @@ You can read more about the Issuer resource in the :doc:`Issuer reference docs <
          name: letsencrypt-staging
 
        # ACME DNS-01 provider configurations
-       dns01:
-
-         # Here we define a list of DNS-01 providers that can solve DNS challenges
-         providers:
-
-           - name: prod-dns
-             clouddns:
-               # A secretKeyRef to a google cloud json service account
-               serviceAccountSecretRef:
-                 name: clouddns-service-account
-                 key: service-account.json
-               # The project in which to update the DNS zone
-               project: gcloud-prod-project
-
-           - name: cf-dns
-             cloudflare:
-               email: user@example.com
-               # A secretKeyRef to a cloudflare api key
-               apiKeySecretRef:
-                 name: cloudflare-api-key
-                 key: api-key.txt
+       solvers:
+       - dns01:
+           # Here we define a list of DNS-01 providers that can solve DNS challenges
+           # Adjust the configuration below according to your environment.
+           # You can view more example configurations for different DNS01
+           # providers in the documentation: https://docs.cert-manager.io/en/latest/tasks/issuers/setup-acme/dns01/index.html
+           cloudflare:
+              email: my-cloudflare-acc@example.com
+              # !! Remember to create a k8s secret before
+              # kubectl create secret generic cloudflare-api-key
+              apiKeySecretRef:
+                name: cloudflare-api-key-secret
+                key: api-key
+           clouddns:
+              # The ID of the GCP project
+              # reference: https://docs.cert-manager.io/en/latest/tasks/issuers/setup-acme/dns01/google.html
+              project: $PROJECT_ID
+              # This is the secret used to access the service account
+              serviceAccountSecretRef:
+                name: clouddns-dns01-solver-svc-acct
+                key: key.json
+           dnsNames:
+           - foo.com
```

Comment:
This solver config isn't actually valid - I've updated it to fix it up:

```yaml
        # Here we define a list of 'solver' providers that can solve ACME challenges.
        # We configure a 'default' CloudDNS based DNS01 solver, as well as a
        # specific Cloudflare solver which is used for solving challenges from the
        # DNS name 'foo.com' only.
        # Adjust the configuration below according to your environment.
        # You can view more example configurations for different DNS01
        # providers in the documentation: https://docs.cert-manager.io/en/latest/tasks/issuers/setup-acme/dns01/index.html
        solvers:
        # An empty 'selector' means that this solver matches all domains
        - selector: {}
          dns01:
            clouddns:
              # The ID of the GCP project
              # reference: https://docs.cert-manager.io/en/latest/tasks/issuers/setup-acme/dns01/google.html
              project: $PROJECT_ID
              # This is the secret used to access the service account
              serviceAccountSecretRef:
                name: clouddns-dns01-solver-svc-acct
                key: key.json

        # We only use cloudflare to solve challenges for foo.com.
        # Alternative options such as 'matchLabels' and 'dnsZones' can be specified
        # as part of a solver's selector too.
        - selector:
            dnsNames:
            - foo.com
          dns01:
            cloudflare:
              email: my-cloudflare-acc@example.com
              # !! Remember to create a k8s secret before
              # kubectl create secret generic cloudflare-api-key
              apiKeySecretRef:
                name: cloudflare-api-key-secret
                key: api-key
```

Reply:
thx a lot